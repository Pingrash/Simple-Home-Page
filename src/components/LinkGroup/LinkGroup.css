.link-container {
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  flex: 1;
  padding: 2rem 6rem;
}
