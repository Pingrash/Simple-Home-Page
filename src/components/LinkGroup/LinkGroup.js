import React from 'react';
import PropTypes from 'prop-types';
import './LinkGroup.css';

import Link from '../Link/Link';

const LinkGroup = ({ links }) => {
  const linksToRender = links.map(link => {
    return (
      <Link
        key={link.link}
        link={link.link}
        favicon={link.favicon}
        data-test='link'
      />
    );
  });

  return (
    <div className='link-container' data-test='link-group'>
      {linksToRender}
    </div>
  );
};

LinkGroup.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      link: PropTypes.string.isRequired,
      favicon: PropTypes.string
    })
  ).isRequired
};

export default LinkGroup;
