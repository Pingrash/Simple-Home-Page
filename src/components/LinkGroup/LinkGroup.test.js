import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { findByDataTest, checkProps } from '../../../Utilities';
import LinkGroup from './LinkGroup';

const setup = (props = {}) => {
  const component = shallow(<LinkGroup {...props} />);
  return component;
};

const props = {
  links: [
    { link: 'link1', favicon: 'favicon1' },
    { link: 'link2', favicon: 'favicon2' },
    { link: 'link3', favicon: 'favicon3' }
  ]
};

describe('Snapshot', () => {
  it('Should render correctly', () => {
    const tree = renderer.create(<LinkGroup {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('LinkGroup component renders correctly', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup(props);
  });

  it('Should render without errors', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Should render three links', () => {
    const links = findByDataTest(wrapper, 'link');
    expect(links.length).toBe(3);
  });
});

describe('LinkGroup component has correct props', () => {
  it('Should not have any props errors', () => {
    const expectedProps = {
      links: [
        { link: 'link1', favicon: 'favicon1' },
        { link: 'link2' }
      ]
    };

    const propsError = checkProps(LinkGroup, expectedProps);

    expect(propsError).toBeUndefined();
  });
});
