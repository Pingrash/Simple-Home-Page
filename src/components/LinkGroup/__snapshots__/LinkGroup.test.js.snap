// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Should render correctly 1`] = `
<div
  className="link-container"
  data-test="link-group"
>
  <a
    className="link-button"
    data-test="link-button"
    href="link1"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="link-title"
    />
    <img
      alt="favicon"
      className="link-favicon"
      data-test="favicon"
      src="favicon1"
    />
  </a>
  <a
    className="link-button"
    data-test="link-button"
    href="link2"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="link-title"
    />
    <img
      alt="favicon"
      className="link-favicon"
      data-test="favicon"
      src="favicon2"
    />
  </a>
  <a
    className="link-button"
    data-test="link-button"
    href="link3"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="link-title"
    />
    <img
      alt="favicon"
      className="link-favicon"
      data-test="favicon"
      src="favicon3"
    />
  </a>
</div>
`;
